#3 Create Checkout Flow

kagaba-etienne opened the pull request:
#### What does this PR do?
- Creates and order after checkout
- Calculates discount from applied coupons
#### Description of Task to be completed

> The backend needs to handle the checkout process, including validating and processing the order, updating inventory, and updating the customer's account.

#### How should this be manually tested?
- Checkout in branch `ft-checktout-3` and pull most recent changes
- Start the server by running `npm run start`
- Make sure you have a user account already, and few products in your cart before proceeding checkout.
- Make a POST request to this route `/api/checkout` with a requestBody containing the following properties `country, street_address, town, state, post_code, email` AKA billing information. Wait for a successful response. Kudos !!! You have just initiated an order.

#### Relevant pivotal tracker stories?
- [Card #3](https://trello.com/c/LUfLttI6)

bryan936M reviewed (approved):
Nice!

yobuir reviewed (approved):
nice work #resolve_conflicts

chanceDira committed:
feature(create checkout flow): a user should be able to check out
- create a checkout route to handle checkout.
- updates inventory after a successful order.

NTElissa reviewed (commented):
Good progress

kagaba-etienne committed:
Merge branch 'develop' into ft-checkout-3

kagaba-etienne committed:
Merge branch 'develop' into ft-checkout-3

chanceDira commented on src/database/models/index.js:
```diff
@@ -13,14 +13,27 @@ const env = _env.NODE_ENV || 'development';
 
-if (config.use_env_variable) {
-  sequelize = new Sequelize(_env[config.use_env_variable], config);
+// if (config.use_env_variable) {
```
Remove these comments

chanceDira commented on tests/Product.test.js:
```diff
@@ -83,4 +88,19 @@ describe('Product API', function() {
         });
     });
   });
+  // describe('DELETE api/product/:id', function() {
```
Remove these comments too

chanceDira reviewed (changes requested):
Good job @kagaba-etienne just one thing

chanceDira reviewed (approved):
Good job @kagaba-etienne
